packages/react/src/components/TextField/TextField.types.ts

```diff
@@ -109,6 +109,15 @@ export interface ITextFieldProps extends React.AllHTMLAttributes<HTMLInputElemen
    */
   onRenderDescription?: IRenderFunction<ITextFieldProps>;
 
+  /**
+   * Custom renderer for input element.
+   */
+  onRenderInput?: (
+    props?: ITextFieldProps,
```

Comment:
I just realized what the more standard solution to this typing issue would be... `IRenderFunction<ITextFieldProps & { inputIds: ITextFieldInputIds }` 🤦‍♀️

Reply:
Or `IRenderFunction<ITextFieldProps & { inputId: string; descriptionId: string; }>`

Reply:
Just made the typing change 😄